collect_app/src/main/java/org/odk/collect/android/formmanagement/ServerFormUseCases.kt

```diff
@@ -88,63 +88,102 @@ object ServerFormUseCases {
 
             val tempMediaFile = File(tempMediaDir, mediaFile.filename)
 
-            val existingFile = searchForExistingMediaFile(formsRepository, formToDownload, mediaFile)
-            existingFile.also {
-                if (it != null) {
-                    if (it.getMd5Hash().contentEquals(mediaFile.hash)) {
-                        FileUtils.copyFile(it, tempMediaFile)
-                    } else {
-                        val existingFileHash = it.getMd5Hash()
-                        val file = formSource.fetchMediaFile(mediaFile.downloadUrl)
-                        FileUtils.interuptablyWriteFile(file, tempMediaFile, tempDir, stateListener)
-
-                        if (!tempMediaFile.getMd5Hash().contentEquals(existingFileHash)) {
-                            newAttachmentsDownloaded = true
-                        }
-                    }
-                } else {
-                    val file = formSource.fetchMediaFile(mediaFile.downloadUrl)
-                    FileUtils.interuptablyWriteFile(file, tempMediaFile, tempDir, stateListener)
+            if (mediaFile.type != null) {
```

Comment:
Can we have an ext function or something that would be like `isEntityFile`? It is confusing that `type != null` means it is a file with entities.